src/tasks/decode.ts

```diff
@@ -0,0 +1,333 @@
+  const columnWidth = displayTokensColumnWidth(formattedTokens);
+  console.log(chalk.bold("=== Decoded tokens ==="));
+  console.log(
+    headers
+      .map((header) => chalk.cyan(cellText(header, columnWidth[header])))
+      .join(chalk.gray("|")),
+  );
+  console.log(
+    chalk.gray(headers.map((key) => "-".repeat(columnWidth[key])).join("+")),
+  );
+  for (const token of formattedTokens) {
+    console.log(
+      headers
+        .map((key) => cellText(token[key], columnWidth[key]))
+        .join(chalk.gray("|")),
+    );
+  }
+  console.log();
+}
+
+function displayTrades(trades: Trade[], tokens: Token[]) {
+  console.log(chalk.bold("=== Decoded trades ==="));
+  console.log(chalk.gray("-".repeat(WIDTH)));
+  for (const trade of trades) {
+    displayTrade(trade, tokens);
+    console.log();
+  }
+}
+
+function formatSignature(sig: SigningScheme): string {
+  switch (sig) {
+    case SigningScheme.EIP712:
+      return "eip-712";
+    case SigningScheme.ETHSIGN:
+      return "ethsign";
+    case SigningScheme.EIP1271:
+      return "eip-1271";
+    case SigningScheme.PRESIGN:
+      return "presign";
+    default:
+      return `invalid (${sig})`;
+  }
+}
+
+function displayTrade(trade: Trade, tokens: Token[]) {
+  const prettyToken = (token: MaybeToken, checkNative?: boolean) =>
+    `${
+      checkNative && token.nativeFlag
+        ? NATIVE_TOKEN
+        : token.symbol ?? INVALID_TOKEN
+    } (${token.index})`;
+  const prettyAmount = (
+    amount: BigNumberish,
+    token: MaybeToken,
+    checkNative?: boolean,
+  ) =>
+    `${utils.formatUnits(amount, token.decimals ?? undefined)} ${prettyToken(
+      token,
+      checkNative,
+    )}`;
+  const {
+    executedAmount,
+    validTo,
+    appData,
+    receiver,
+    sellAmount,
+    buyAmount,
+    feeAmount,
+    sellTokenIndex,
+    buyTokenIndex,
+    flags,
+    signature,
+  } = trade;
+  const { kind, partiallyFillable, signingScheme } = decodeTradeFlags(flags);
+  const sellToken = tokens[sellTokenIndex] ?? { index: sellTokenIndex };
+  const buyToken = tokens[buyTokenIndex] ?? { index: buyTokenIndex };
+  console.log(
+    mainLabel("Order"),
+    `${kind.toString().toUpperCase()} ${
+      partiallyFillable ? "partially fillable " : ""
+    }order, valid until ${new Date(validTo * 1000).toISOString()} (${validTo})`,
+  );
+  console.log(label(`Trade`), `sell ${prettyAmount(sellAmount, sellToken)}`);
+  console.log("      ", ` buy ${prettyAmount(buyAmount, buyToken, true)}`);
+  console.log("      ", ` fee ${prettyAmount(feeAmount, sellToken)}`);
+  if (partiallyFillable) {
+    console.log(
+      label(`Executed amount`),
+      `${prettyAmount(executedAmount, sellToken)}`,
+    );
+  }
+  if (receiver !== constants.AddressZero) {
+    console.log(label(`Receiver`), receiver);
+  }
+  if (appData !== constants.HashZero) {
+    console.log(label(`AppData`), appData);
+  }
+  console.log(
+    label(`Signature (${formatSignature(signingScheme)})`),
```

Comment:
If possible, it would be great to also print owner address and order UID here.

Reply:
Implemented in #598.